StormDBManager.py

```diff
@@ -0,0 +1,300 @@
+        If it fails, the _version field remains at 0 as defined in the init function.
+        """
+        try:
+            version_str, = yield self.fetchone(u"SELECT value FROM MyInfo WHERE entry == 'version'")
+            self._version = int(version_str)
+            self._logger.info(u"Current database version is %s", self._version)
+        except (TypeError, OperationalError):
+            self._logger.warning(u"Failed to load database version, setting the DB version to 0.")
+            self._version = 0
+
+    def schedule_query(self, callable, *args, **kwargs):
+        """
+        Utility function to schedule a query to be executed using the db_lock.
+
+        Args:
+            callable: The database function that is to be executed.
+            *args: Any additional arguments that will be passed as the callable's arguments.
+            **kwargs: Keyword arguments that are passed to the callable function.
+
+        Returns: A deferred that fires with the result of the query.
+
+        """
+        return self.db_lock.run(callable, *args, **kwargs)
+
+    def execute(self, query, arguments=None, get_lastrowid=False):
+        """
+        Executes a query on the twisted thread pool using the Storm framework.
+
+        Args:
+            query: The sql query to be executed.
+            arguments: Optional arguments that go with the sql query.
+            get_lastrowid: If true, this function will return the last inserted row id, otherwise None.
+
+        Returns: A deferred that fires once the execution is done, the result will be None if get_lastrowid is False
+        else it returns with the last inserted row id.
+
+        """
+
+        @transact
+        def _execute(self, query, arguments=None, get_lastrowid=False):
+            connection = Connection(self._database)
+            ret = None
+            if get_lastrowid:
+                result = connection.execute(query, arguments, noresult=False)
+                ret = result._raw_cursor.lastrowid
+            else:
+                connection.execute(query, arguments, noresult=True)
+            connection.close()
+            return ret
+
+        return self.db_lock.run(_execute, self, query, arguments, get_lastrowid)
+
+    def executemany(self, query, list):
+        """
+        Executes a query on the twisted thread pool using the Storm framework many times using the values provided by
+        a list.
+
+        Args:
+            query: The sql query to be executed.
+            list: The list containing tuples of values to execute the query with.
+
+        Returns: A deferred that fires once the execution is done, the result will be None.
+
+        """
+        def _execute(connection, query, arguments=None):
+            connection.execute(query, arguments, noresult=True)
+
+        @transact
+        def _executemany(self, query, list):
+            connection = Connection(self._database)
+            for item in list:
```

Comment:
The `list` variable name shadows the built-in list keyword. Maybe rename it? (there are more occurrences where you use `list` as variable name iirc).